source/base/parameter_handler.cc

```diff
@@ -1040,14 +1037,52 @@ ParameterHandler::set(const std::string &entry_string, const bool new_value)
   set(entry_string, (new_value ? "true" : "false"));
 }
 
+void
+ParameterHandler::recursively_sort_parameters(
+  const std::vector<std::string> &target_subsection_path) const
+{
+  boost::property_tree::ptree &current_section =
+    entries->get_child(collate_path_string(target_subsection_path));
 
+  current_section.sort();
+
+  // Now transverse subsections tree recursively.
+  for (auto &p : current_section)
+    {
+      if ((is_parameter_node(p.second) == false) &&
+          (is_alias_node(p.second) == false))
+        {
+          const std::string subsection = demangle(p.first);
+
+          std::vector<std::string> subsection_path = target_subsection_path;
+          subsection_path.emplace_back(subsection);
+
+          recursively_sort_parameters(subsection_path);
+        }
+    }
+}
 
 std::ostream &
 ParameterHandler::print_parameters(std::ostream &    out,
-                                   const OutputStyle style) const
+                                   const OutputStyle style,
+                                   const bool        sort_alphabetical) const
 {
   AssertThrow(out, ExcIO());
 
+  // Save backup of entries before sorting, so that
+  // it can be restored by the end of this method.
+  boost::property_tree::ptree entries_backup;
+
+  // Sort parameters alphabetically, if needed.
+  if (sort_alphabetical)
+    {
+      entries_backup = (*entries);
```

Comment:
Can you sort the backup and print the backup instead? This way you are not modifying the data structures stored at all.

Reply:
Done @tjhei! I've modified a few methods to make them modify the backup copy instead of the class state.